valqueries-automapper/src/main/java/com/valqueries/automapper/ValqueriesAccessDataLayerImpl.java

```diff
@@ -175,25 +176,62 @@ public CrudUpdateResult save(T t) {
 		return getUpdateResult(tx.update(sql, columnizer));
 	}
 
+	private <O> CrudUpdateResult insertInternal(ITransactionContext tx, Collection<O> ts, Class<O> oClass) throws ValqueriesDuplicateKeyException {
+		if (ts.isEmpty()) {
+			return () -> 0;
+		}
+		CompoundColumnizer<O> columnizer = new CompoundColumnizer<O>(genericFactory, mappingHelper, ts, sqlNameFormatter);
+		String sql = dialect.getInsert(columnizer, oClass);
+		UpdateResult result;
+		try {
+			result = tx.update(sql, columnizer); 
+		} catch (Exception e){
+			// TODo should we restrict the exceptions that we flag as duplicate key ?
+			throw new ValqueriesDuplicateKeyException(e);
```

Comment:
How do we know that this is a duplicate key problem, and not e.g. a transaction conflict?

Reply:
We don't, hence the TODO comment. I will change the exception to be more generic, the reason shouldn't matter too much anyway.

Reply:
After talking to Anton, I tried to see if we could catch some specific jdbc exceptions only. However, the SQL Server exceptions do not inherit the jdbc ones, making it impossible to catch in the repository layer. To avoid any big refactoring, I just generalized the exception, as it makes sense to tell the outer layer that the insert fail regardless of the reason.